lib/mumuki/classroom/event/progress_transfer/base.rb

```diff
@@ -0,0 +1,112 @@
+class Mumuki::Classroom::Event::ProgressTransfer::Base
+  attr_reader :indicator, :source_organization
+
+  def initialize(indicator, source_organization, destination_organization)
+    @indicator = indicator
+    @source_organization = source_organization
+    @destination_organization = destination_organization
+  end
+
+  def execute!
+    raise ActiveRecord::RecordNotFound, "Mumuki::Classroom::Student not found" unless old_student && new_student
+
+    new_student.destroy_progress_for_guide!(indicator.content)
```

Comment:
Is there to implement a pseudo-transaction in our current version? It would be nice to avoid deleting old progress until we make sure that everything worked as expected

Reply:
we talked about this over call, but I wanna leave the answer here too:

By default, mongodb provides single document, single write operation atomicity.
Starting in mongodb 3.6, it also provides two-phase commits allowing for single document, multiple write operations atomicty. (https://docs.mongodb.com/v3.6/core/write-operations-atomicity/)
Lastly, mutli-document transactions support was added starting in mongodb 4.0. (https://docs.mongodb.com/v4.0/core/write-operations-atomicity/)